src/tokens/TokenDetails.tsx

```diff
@@ -92,13 +96,23 @@ export default function TokenDetailsScreen({ route }: Props) {
           errorFallback={(localCurrencySymbol ?? '$').concat(' --')}
         />
         {!token.isStableCoin && <PriceInfo token={token} />}
-        {token.isNative && token.symbol === 'CELO' && (
-          <CeloGoldHistoryChart
-            color={Colors.black}
+        {token.isNative && usePriceHistoryFromBlockchainApi ? (
+          <PriceHistoryChart
+            tokenId={tokenId}
             containerStyle={styles.chartContainer}
             chartPadding={Spacing.Thick24}
-            testID="TokenDetails/Chart"
+            testID={`TokenDetails/Chart/${tokenId}`}
+            color={Colors.black}
           />
+        ) : (
+          token.symbol === 'CELO' && (
```

Comment:
we can also switch this to use `networkConfig.celoTokenId`

Reply:
Fixed in c138a17891cf98987f197d691a48ea1e5890a04e!